en: add dmcluster usage document (#792)
pingcap/docs-tidb-operator: en/enable-tls-for-dm.md

- Create the `dm-cluster.yaml` file:

```yaml
apiVersion: pingcap.com/v1alpha1
kind: DMCluster
metadata:
  name: ${cluster_name}
  namespace: ${namespace}
spec:
  tlsCluster:
    enabled: true
  version: v2.0.0-rc.2
  pvReclaimPolicy: Retain
  discovery:
    address: "http://${tidb_cluster_name}-discovery.${tidb_namespace}:10261"
  master:
    baseImage: pingcap/dm
    replicas: 1
    storageSize: "1Gi"
    config:
      cert-allowed-cn:
        - TiDB
  worker:
    baseImage: pingcap/dm
    replicas: 1
    storageSize: "1Gi"
    config:
      cert-allowed-cn:
        - TiDB
```

Use the `kubectl apply -f dm-cluster.yaml` file to create a DM cluster.

### Configure `dmctl` and connect to the cluster

Get into the DM-master Pod:

{{< copyable "shell-regular" >}}

``` shell
kubectl exec -it ${cluster_name}-dm-master-0 -n ${namespace} sh
```

Use `dmctl`:

{{< copyable "shell-regular" >}}

``` shell
cd /var/lib/dm-master-tls
/dmctl --ssl-ca=ca.crt --ssl-cert=tls.crt --ssl-key=tls.key --master-addr https://127.0.0.1:8261 list-member
```

> ## Enable TLS-authenticated MySQL/TiDB database of MySQL client

Reviewer: ```suggestion
## Use DM to migrate data between MySQL/TiDB databases that enable TLS for the MySQL client
```

- Create the `dm-cluster.yaml` file:

```yaml
apiVersion: pingcap.com/v1alpha1
kind: DMCluster
metadata:
  name: ${cluster_name}
  namespace: ${namespace}
spec:
  tlsCluster:
    enabled: true
  version: v2.0.0-rc.2
  pvReclaimPolicy: Retain
  discovery:
    address: "http://${tidb_cluster_name}-discovery.${tidb_namespace}:10261"
  master:
    baseImage: pingcap/dm
    replicas: 1
    storageSize: "1Gi"
    config:
      cert-allowed-cn:
        - TiDB
  worker:
    baseImage: pingcap/dm
    replicas: 1
    storageSize: "1Gi"
    config:
      cert-allowed-cn:
        - TiDB
```

Use the `kubectl apply -f dm-cluster.yaml` file to create a DM cluster.

### Configure `dmctl` and connect to the cluster

Get into the DM-master Pod:

{{< copyable "shell-regular" >}}

``` shell
kubectl exec -it ${cluster_name}-dm-master-0 -n ${namespace} sh
```

Use `dmctl`:

{{< copyable "shell-regular" >}}

``` shell
cd /var/lib/dm-master-tls
/dmctl --ssl-ca=ca.crt --ssl-cert=tls.crt --ssl-key=tls.key --master-addr https://127.0.0.1:8261 list-member
```

## Use DM to migrate data between MySQL/TiDB databases that enable TLS for the MySQL client

> The following section describes how to configure DM to enable synchronously TLS-authenticated MySQL/TiDB database of MySQL client.

Reviewer: ```suggestion
This section describes how to configure DM to migrate data between MySQL/TiDB databases that enable TLS for the MySQL client.

To learn how to enable TLS for the MySQL client of TiDB, refer to [Enable TLS for the MySQL Client](enable-tls-for-mysql-client.md).

> ### Step 1: Create the Kubernetes Secret object of TLS of each MySQL client

Reviewer: ```suggestion
### Step 1: Create the Kubernetes Secret object for each TLS-enabled MySQL client
```
@lichunzhu PTAL if the translation is correct.

Reviewer 2: ```suggestion
### Step 1: Create the Kubernetes Secret object for each TLS-enabled MySQL
```

> Suppose you deployed a MySQL/TiDB database with MySQL client TLS enabled. To create a Secret object for the TiDB cluster using the following command:

Reviewer: ```suggestion
Suppose you have deployed a MySQL/TiDB database with TLS-enabled for the MySQL client. To create Secret objects for the TiDB cluster, execute the following command:
```
It seems the commands below create two objects.

{{< copyable "shell-regular" >}}

```shell
kubectl create secret generic ${mysql_secret_name1} --namespace=${namespace} --from-file=tls.crt=client.pem --from-file=tls.key=client-key.pem --from-file=ca.crt=ca.pem
kubectl create secret generic ${tidb_secret_name} --namespace=${namespace} --from-file=tls.crt=client.pem --from-file=tls.key=client-key.pem --from-file=ca.crt=ca.pem
```

> ### Step 2: Mount secret objects to DM cluster

Reviewer: ```suggestion
### Step 2: Mount the Secret objects to the DM cluster
```

> After creating the Kubernetes Secret object of the upstream and downstream databases, you need to set `spec.tlsClientSecretNames` so that you can mount the Secret object to the Pod of DM-master/DM-worker.

Reviewer: ```suggestion
After creating the Kubernetes Secret objects for the upstream and downstream databases, you need to set `spec.tlsClientSecretNames` so that you can mount the Secret objects to the Pod of DM-master/DM-worker.
```

```yaml
apiVersion: pingcap.com/v1alpha1
kind: DMCluster
metadata:
  name: ${cluster_name}
  namespace: ${namespace}
spec:
  version: v2.0.0-rc.2
  pvReclaimPolicy: Retain
  discovery:
    address: "http://${tidb_cluster_name}-discovery.${tidb_namespace}:10261"
  tlsClientSecretNames:
    - ${mysql_secret_name1}
    - ${tidb_secret_name}
  master:
    ...
```

### Step 3: Modify the data source and migration task configuration

> After setting the `spec.tlsClientSecretNames` option, TiDB Operator will mount the Secret object ${secret_name} to the path `/var/lib/source-tls/${secret_name}`.

Reviewer: ```suggestion
After configuring `spec.tlsClientSecretNames`, TiDB Operator will mount the Secret objects `${secret_name}` to the path `/var/lib/source-tls/${secret_name}`.
```

> 1. Fill in the [data source configuration](deploy-tidb-dm.md#create-data-source) `from.security` option of `source1.yaml` file:

Reviewer: ```suggestion
1. Configure `from.security` in the `source1.yaml` file as described in the [data source configuration](deploy-tidb-dm.md#create-data-source):
```

    ``` yaml
    source-id: mysql-replica-01
    from:
      host: ${mysql_host1}
      user: dm
      password: ""
      port: 3306
      security:
        ssl-ca: /var/lib/source-tls/${mysql_secret_name1}/ca.crt
        ssl-cert: /var/lib/source-tls/${mysql_secret_name1}/tls.crt
        ssl-key: /var/lib/source-tls/${mysql_secret_name1}/tls.key
    ```

> 2. Fill in the `target-database.security` option of the `task.yaml` file in the [Configure Migration Tasks](deploy-tidb-dm.md#configure-migration-tasks):

Reviewer: ```suggestion
2. Configure `target-database.security` in the `task.yaml` file as described in the [Configure Migration Tasks](deploy-tidb-dm.md#configure-migration-tasks):
```
